content/articles/how-to-build-authentication-api-with-jwt-token-in-nodejs/index.md

```diff
@@ -0,0 +1,499 @@
+
+#### Authorization
+Authorization is the process of allowing authenticated users to access resources by determining whether they have system access permissions. By giving or denying specific licenses to an authenticated user, authorization enables you to control access privileges.
+
+So, authorization occurs after the system authenticates your identity, granting you complete access to resources such as information, files, databases, funds, places, and anything else. So said, authorization affects your capacity to access the system and the extent to which you can do so.
+
+### What is JWT
+JSON Web Tokens are an RFC 7519 open industry standard for representing claims between two parties. You can use [jwt.io](https://jwt.io) to decode, verify, and produce JWT.
+
+JSON Web Token (JWT) specifies a compact and self-contained method for communicating information as a JSON object between two parties. Because it is signed, this information can be checked and trusted. JWTs can be signed using a secret (using the HMAC algorithm) or an RSA or ECDSA public/private key combination. In a moment, we'll see some instances of how to use them.
+
+#### Prerequisites
+- A working knowledge of Javascript.
+- A good understanding of NodeJS is required.
+- A basic understanding of MongoDB or any Database of your choice.
+
+### API development using JWT token for authentication in NodeJS
+To get started, we need to set up our project.
+
+#### Step 1 - Open your IDE or code editor
+
+To open Visual Studio Code, navigate to any directory of your choice on your machine. In the terminal, execute:
+
+```bash
+code.
+```
+
+> **Note**: `code .` won't work if you don't have Visual Studio Code installed on your system.
+
+#### Step 2 - Create a directory and initialize `npm`
+
+Create a directory and initialize npm by typing the following command:
+
+- Windows power shell
+
+```bash
+mkdir jwt-project
+cd jwt-project
+npm init -y
+```
+
+- Linux
+
+```bash
+mkdir jwt-project
+cd jwt-project
+npm init -y
+```
+
+#### Step 3 - Create files and directories as shown below
+![folder structure](/engineering-education/how-to-build-authentication-api-with-jwt-token-in-nodejs/folder-struc.png)
+
+#### Step 4 - Install dependencies
+We'll install several dependencies like `mongoose`, `jsonwebtoken`, `express` `dotenv` `bcryptjs` and development dependency like `nodemon` to restart the server as we make changes automatically.
+
+We will install mongoose because I will be using MongoDB for storage while we validate user credentials against what we have in our database. The whole authentication process is not limited to the database we'll be using in this article.
+
+```bash
+npm install mongoose express jsonwebtoken dotenv bcryptjs
+npm install nodemon -D
+```
+
+#### Step 5 - Create a NodeJS server and connect your database
+
```

Comment:
Provide a brief explanation of the code in this section.

Reply:
ok